sdk/core/Azure.Core.Extensions/src/Internal/ClientOptionsMonitor.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+
+namespace Azure.Core.Extensions
+{
+    // Slightly adjusted copy of https://github.com/aspnet/Extensions/blob/master/src/Options/Options/src/OptionsMonitor.cs
+    internal class ClientOptionsMonitor<TClient, TOptions> : IOptionsMonitor<TOptions>, IDisposable where TOptions : class
```

Comment:
How does this relate to adding ServiceVersion?

Reply:
I had to copy a bit of ASP.NET Core options infrastructure because by default it doesn't support parametrized option type constructors (which ClientOPtions ctor are)